selfhost/typechecker.jakt

```diff
@@ -2719,7 +2719,7 @@ struct Typechecker {
                     is_instantiated: false
                     parsed_function: None
                     is_comptime: false
-                    is_closure: false
+                    is_closure: true
```

Comment:
Isn't this a function expression without any captures? that's not what this flag is for.
Maybe replace this field to be an enum?

Reply:
Ah, I have updated it to add to `FunctionType`
`Expression` and `Closure`
Let me know if that is okay, it seems that there are no `is_closure=true` cases currently so I am not sure that it is fine for that case.

Reply:
pinging @alimpfard - does the above fix the issue you saw?

Reply:
Looks right, sorry I forgot about this one :sweat_smile: